packages/n/niri/package.yml

```diff
@@ -0,0 +1,46 @@
+name       : niri
+version    : 25.05.1
+release    : 1
+source     :
+    - https://github.com/YaLTeR/niri/archive/refs/tags/v25.05.1.tar.gz : 2df192fa85adbb0eabcfddbd8aa812c3d3329a582a7e62212ea9fcddafa79b00
+homepage   : https://github.com/YaLTeR/niri
+license    : GPL-3.0-or-later
+component  : desktop
+summary    :
+    - session : Session files for Niri
+    - A scrollable-tiling Wayland compositor
+description:
+    - session : Session files for Niri.
+    - A scrollable-tiling Wayland compositor.
+builddeps  :
+    - pkgconfig(gbm)
+    - pkgconfig(libdisplay-info)
+    - pkgconfig(libinput)
+    - pkgconfig(libseat)
+    - pkgconfig(pangocairo)
+    - pkgconfig(xkbcommon)
+    - rust
+rundeps    :
+    - session :
+        - niri
```

Comment:
We also need the following because of the contents of `niri-portals.conf`:
- xdg-desktop-portal-gtk
- xdg-desktop-portal-gnome

Reply:
I really don't want to add a rundep on xdp-gnome, because it breaks any other portal if they're installed.

Reply:
Then at the minimum `gnome-keyring` should be added: https://github.com/YaLTeR/niri/blob/e0b0b04b445f7044f383e50104f861e632e1c905/resources/niri-portals.conf#L5

> because it breaks any other portal if they're installed

Interesting, I have xdp-gnome, xdp-gtk, and xdg-wlr all installed and I've not had issues so far.

Reply:
The provided `niri-portals.conf` prefers to use xdg-gnome first then fallbacks to xdg-gtk, so if we don't have xdg-gnome installed, to my knowledge at least the following will break:
1. Filepicker dialogue.
2. Screencasting support (e.g. google meet screenshare)

So we should either patch `niri-portals.conf` or add xdg-gnome to our rundep. Exactly how we should patch `niri-portals.conf` I'm not sure. I just have xdp-gnome installed along with other portals and so far it works fine :eyes: